.github/workflows/publish-rs.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish canbench-rs to crates.io
+
+on:
+  workflow_run:
+    workflows: [Publish canbench-rs-macros to crates.io]
+    types:
+      - completed
+
+jobs:
+  publish-canbench-rs:
+    runs-on: ubuntu-latest
+    
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    
+    permissions:
+      id-token: write # Required for OIDC token exchange
+
+    steps:
+      - uses: actions/checkout@v5
+
+      - uses: rust-lang/crates-io-auth-action@v1
+        id: auth
+
+      - name: Set Cargo.toml versions
+        shell: bash
+        env:
+          RELEASE_TAG: ${{ github.ref }}
```

Comment:
Just realized that this will not work for the workflows triggered by completion of the previous workflow. I guess we'd either somehow have to directly access the tag or just combine the publish workflows into one. I'm currently more in favour of the latter as the split into multiple workflows adds quite some unnecessary duplication and is not needed as all crates are published at the same time anyways.